src/foam/u2/CitationView.js

```diff
@@ -34,7 +34,17 @@ foam.CLASS({
       name: 'updateSummary',
       isFramed: true,
       code: function() {
-        this.summary = this.data ? this.data.toSummary() : undefined;
+        if ( this.data ) {
```

Comment:
what if toSummary is async and we await in it, so each model can decide what to do and we don't have two methods and in the case of transactions - they do completely different things.